spark-jobs/src/test/scala/za/co/absa/enceladus/common/version/SparkVersionGuardSuite.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common.version
+
+import org.scalatest.{FlatSpec, Matchers}
+import za.co.absa.commons.version.Version._
+
+class SparkVersionGuardSuite extends FlatSpec with Matchers {
+
+  "SparkVersionGuard" should "check basic version compatibility" in {
+    SparkVersionGuard(semver"2.4.4", semver"3.0.0").checkSparkVersionCompatibility(semver"2.4.5")
+    SparkVersionGuard(semver"2.4.4", semver"3.0.0").checkSparkVersionCompatibility(semver"2.4.4") // min is inclusive
+    SparkVersionGuard(semver"1.6.0", semver"3.0.0").checkSparkVersionCompatibility(semver"2.4.4")
+
+    assertThrows[AssertionError] {
```

Comment:
Could you add a check for the message thrown as well?

Reply:
Done, I am checking that the part of the `SparkVersionGuard` message appears to make sure that the source of the exception is what is expected.